packages/block-library/src/block/edit.js

```diff
@@ -401,47 +215,16 @@ function ReusableBlockEdit( {
 		),
 	} );
 
+	// Use `blocks` variable until `innerBlocks` is populated, which has the proper clientIds.
 	const innerBlocksProps = useInnerBlocksProps( blockProps, {
 		templateLock: 'contentOnly',
 		layout,
-		renderAppender: innerBlocks?.length
-			? undefined
-			: InnerBlocks.ButtonBlockAppender,
+		value: innerBlocks.length > 0 ? innerBlocks : blocks,
+		onInput: NOOP,
+		onChange: NOOP,
+		renderAppender: blocks?.length ? undefined : blocks.ButtonBlockAppender,
```

Comment:
Is this a typo?

Reply:
(`blocks.ButtonBlockAppender` will never exist)

Reply:
It seems that change has been there since the [first commit](https://github.com/WordPress/gutenberg/pull/60721/commits/78f960185df3dc1c3d26f084f9375b4431444fab), so I assume we just replaced innerBlocks by blocks. Maybe we should go back to the previous logic?

```
renderAppender: innerBlocks?.length
    ? undefined
    : InnerBlocks.ButtonBlockAppender,
```